fuse_core/include/fuse_core/local_parameterization.h

```diff
@@ -86,6 +88,21 @@ class LocalParameterization : public ceres::LocalParameterization
   virtual bool ComputeMinusJacobian(
     const double* x,
     double* jacobian) const = 0;
+
+private:
+  // Allow Boost Serialization access to private methods
+  friend class boost::serialization::access;
+
+  /**
+   * @brief The Boost Serialize method that serializes all of the data members in to/out of the archive
+   *
+   * @param[in/out] archive - The archive object that holds the serialized class members
+   * @param[in] version - The version of the archive being read/written. Generally unused.
+   */
+  template<class Archive>
+  void serialize(Archive& /* archive */, const unsigned int /* version */)
```

Comment:
Why do we need this if there's no attributes to de/serialize?

Reply:
If we serialize an object by base pointer (which is done in the MarginalConstraint class), the base class needs to have a serialize() method. There may be a Boost macro or switch somewhere that makes this less invasive, but I haven't found it.

Reply:
Thanks for clarifying. I was expecting that when I saw the children classes of this one. I'm not familiar enough with Boost serialization, so I don't know if there's a shorter way to do this. Either way, it LGTM.